fleetspeak/src/client/https/polling_test.go

```diff
@@ -427,46 +425,59 @@ func (hp *httpsProxy) close() {
 // Handles the request by hijacking the HTTP connection, opening a second connection to the host of
 // the request and copying data between the two connections.
 func (hp *httpsProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	t := hp.t
+	returnError := func(err error) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+
 	d, err := httputil.DumpRequest(r, false)
 	if err != nil {
-		t.Fatal(err)
+		returnError(err)
+		return
 	}
-	t.Logf("Got proxy request: %s.", string(d))
+	log.Infof("Got proxy request: %s.", string(d))
+
 	if r.Method != http.MethodConnect {
-		t.Fatalf("Proxy received invalid method: %s.", r.Method)
+		http.Error(w, fmt.Sprintf("Proxy received invalid method: %s.", r.Method), http.StatusInternalServerError)
+		return
 	}
 	conn, err := net.Dial("tcp", r.Host)
 	if err != nil {
-		t.Fatal(err)
+		returnError(err)
+		return
 	}
+	defer conn.Close()
 	w.WriteHeader(http.StatusOK)
 	hijacker, ok := w.(http.Hijacker)
 	if !ok {
-		t.Fatal("Failed to hijack HTTP connection.")
+		log.Error("Failed to hijack HTTP connection.")
+		return
 	}
 	httpConn, _, err := hijacker.Hijack()
 	if err != nil {
-		t.Fatal(err)
+		log.Error(err)
+		return
 	}
+	defer httpConn.Close()
 	atomic.AddUint32(&hp.atomicNumRequests, 1)
 	c := make(chan struct{})
 	copyFromTo := func(from net.Conn, to net.Conn) {
 		_, err := io.Copy(to, from)
 		if err != nil {
-			t.Fatal(err)
+			log.Error(err)
```

Comment:
Same here.

This error message will appear among several messages in the test log. Without context, it might not be very helpful.

Reply:
Done.